tests/test_api.py

```diff
@@ -441,6 +503,15 @@ def test_support_for_unrecognized_fields(self):
                     dict1["signed"]["keys"][keyid]["d"] = "c"
                 for role_str in dict1["signed"]["roles"].keys():
                     dict1["signed"]["roles"][role_str]["e"] = "g"
+            elif metadata == "targets" and dict1["signed"].get("delegations"):
+                for keyid in dict1["signed"]["delegations"]["keys"].keys():
+                    dict1["signed"]["delegations"]["keys"][keyid]["d"] = "c"
+                new_roles = []
+                for role in dict1["signed"]["delegations"]["roles"]:
+                    role["e"] = "g"
+                    new_roles.append(role)
+                dict1["signed"]["delegations"]["roles"] = new_roles
+                dict1["signed"]["delegations"]["foo"] = "bar"
```

Comment:
I know this is ugly, but until @jku proposes his test where he automatically validates support for unrecognized fields on each level of the dictionary, we would have to stick to this.

Reply:
the reason I haven't done it is that while there seemingly was a consensus that "unrecognised keys should be allowed everywhere in the file format" ... it's not really true. There's a lot of places where that would be weird and unsafe (what if someone inserts random data into "keys" or "roles") and we should not allow that. The real rule is probably "unreconised keys should be allowed at every _object_ root level": so you should be allowed to insert a key into Signed, Key or Role ... but not into any dictionary where we cannot know the recognised key names before hand (like "keys").

Poisoning everything except a hard-coded list of dictionaries (instead of only poisoning specific dicts) may still be the better testing solution but it's not trivial

Reply:
there's an implementation in #1382 now